bloomfilter/src/Data/BloomFilter/BitVec64.hs

```diff
@@ -31,8 +32,24 @@ unsafeIndex (BV64 bv) i = testBit (P.unsafeIndex bv (w2i j)) (w2i k)
 
 newtype MBitVec64 s = MBV64 (P.MVector s Word64)
 
+-- | Will create an explicitly pinned byte array if it is larger than 1 kB.
+-- This is done because pinned byte arrays allow for more efficient
+-- serialisation, but the definition of 'isByteArrayPinned' changed in GHC 9.6,
+-- see <https://gitlab.haskell.org/ghc/ghc/-/issues/22255>.
+--
+-- TODO: remove this workaround once a solution exists, e.g. a new primop that
+-- allows checking for implicit pinning.
 new :: Word64 -> ST s (MBitVec64 s)
-new s = MBV64 <$> MP.new (w2i (roundUpTo64 s))
+new s
+  | numWords >= 128 = do
+    mba <- newPinnedByteArray (w2i numBytes)
+    return (MBV64 (P.MVector 0 (w2i numWords) mba))
+  | otherwise = do
+    mba <- newByteArray (w2i numBytes)
+    return (MBV64 (P.MVector 0 (w2i numWords) mba))
+  where
+    !numWords = roundUpTo64 s
+    !numBytes = unsafeShiftL numWords 3 -- * 8
```

Comment:
I don't like this, but otherwise we will always need to copy the whole filter to write it to disk. We could decide that cost is acceptable until on a range of GHC versions before having a new primop.